tests/runtimes/android/android_runtime_tests.py

```diff
@@ -0,0 +1,271 @@
+    custom_js_file = os.path.join(APP_NAME, "app", "my-custom-class.js")
+    tns_folder = os.path.join(APP_NAME, 'platforms', 'android/', 'app', 'src', 'main', "java", "com", "tns")
+    gen_folder = os.path.join(tns_folder, "gen")
+    generated_java_file = os.path.join(tns_folder, "MyJavaClass.java")
+
+    @classmethod
+    def setUpClass(cls):
+        TnsTest.setUpClass()
+        cls.emulator = DeviceManager.Emulator.ensure_available(Emulators.DEFAULT)
+        Folder.clean('./' + APP_NAME)
+
+    def tearDown(self):
+        Tns.kill()
+        TnsTest.tearDown(self)
+
+    @classmethod
+    def tearDownClass(cls):
+        TnsTest.tearDownClass()
+        Folder.clean(APP_NAME)
+
+    def test_200_calling_custom_generated_classes_declared_in_manifest(self):
+        Tns.create(app_name=APP_NAME, template=Template.HELLO_WORLD_JS.local_package, update=True)
+        Tns.platform_add_android(APP_NAME, framework_path=Android.FRAMEWORK_PATH)
+        File.copy(os.path.join(TEST_RUN_HOME, 'assets', 'runtime', 'android', 'files',
+                               'calling_custom_generated_classes_declared_in_manifest', 'AndroidManifest.xml'),
+                  os.path.join(TEST_RUN_HOME, APP_NAME, 'app', 'App_Resources', 'Android', 'src', 'main',
+                               'AndroidManifest.xml'))
+        File.copy(os.path.join(TEST_RUN_HOME, 'assets', 'runtime', 'android', 'files',
+                               'calling_custom_generated_classes_declared_in_manifest', 'my-custom-class.js'),
+                  os.path.join(TEST_RUN_HOME, APP_NAME, 'app', 'my-custom-class.js'))
+        File.copy(os.path.join(TEST_RUN_HOME, 'assets', 'runtime', 'android', 'files',
+                               'calling_custom_generated_classes_declared_in_manifest',
+                               'custom-activity.js'),
+                  os.path.join(TEST_RUN_HOME, APP_NAME, 'app', 'custom-activity.js'))
+        Adb.clear_logcat(device_id=self.emulator.id)
+        Tns.run_android(APP_NAME, device=self.emulator.id, just_launch=True, wait=True)
+        Wait.until(lambda: 'we got called from onCreate of custom-activity.js' in Adb.get_logcat(
+            device_id=self.emulator.id))
+        output = Adb.get_logcat(device_id=self.emulator.id)
+
+        # make sure app hasn't crashed
+        assert "Displayed org.nativescript.TNSApp/com.tns.ErrorReportActivity" not in output, \
+            "App crashed with error activity"
+        # check if we got called from custom activity that overrides the default one
+        assert "we got called from onCreate of custom-activity.js" in output, "Expected output not found"
+        # make sure we called custom activity declared in manifest
+        assert "we got called from onCreate of my-custom-class.js" in output, "Expected output not found"
+
+    def test_300_verbose_log_android(self):
+        Tns.create(app_name=APP_NAME, template=Template.HELLO_WORLD_JS.local_package, update=True)
```

Comment:
Can we use Template.MIN_JS?

Reply:
All tests are based on hello world js template. I would prefer using it.